### DISCUSSION TITLE
TRequest.DataSetAdapter

### DISCUSSION THREAD

**Discussion by rssprog** [viniciussanchez/RESTRequest4Delphi#82] [General]
Boa noite, amigos.

Se alguém aqui usa o Lazarus, poderiam por gentileza, me dizer qual a classe utilizada para o TRequest.DataSetAdapter?
Já tentei as classes TBufDataset, TMemDataset e TSQLQuery, mas não obtive sucesso, pois é reportado a exception da classe EJSON.

Obrigado

**Comment by viniciussanchez** [comment]
Bom dia, chegou a olhar a pasta "samples"? Lá tem um exemplo funcionando com Lazarus, talvez te ajude

**Reply by rssprog** [reply]
Olá Vinícius.
Não olhei não, mas vou fazer isso.
Por hora, muito obrigado pelo pronto atendimento.

Forte abraço.

**Reply by IdeaSystems** [reply]
Olá @rssprog!
Estou passando pelo mesmo problema que você usando Lazarus, RESTRequest4D  no SO Ubuntu.
Conseguiu encontrar uma solução para o seu problema?
No meu caso, algumas requisições funcionam e outras não. :/

**Comment by IdeaSystems** [comment]
O meu erro é aleatório, em alguns casos eu consigo usar o .DataSetAdapter(Dataset) em outro uso e da erro mas todos os casos eu recebo o JSON válido como retorno.
Poderia me ajudar @viniciussanchez ?

**Reply by viniciussanchez** [reply]
Se algumas funcionam e outras não, precisa ver o que você está fazendo, como está fazendo... são muitas variáveis em jogo pra poder te ajudar assim

**Reply by IdeaSystems** [reply]
Olá!
Após o erro recebo o retorno do JSON de forma correta como abaixo:

{
  "status": "success",
  "data": {
    "id": 380,
    "fantasia": "Fantasia",
    "cpf_cnpj": "00.000.000/0000-00",
    "logomarca": "https://dominio.nyc3.digitaloceanspaces.com/files/emp380.jpeg",
    "slogan": "slogan",
    "ativo": true,
    "system_user_id": 475
  }
}

O erro acontece no arquivo DataSet.Serialize.Import;
O erro acontece na linha 556 
"Length({$IF DEFINED(FPC)}AJSONObject.Items[I].AsString{$ELSE}LJSONPair.JsonValue.Value{$ENDIF}) > 4096 then"

Estou usando o projeto de exemplo

procedure TFrmMain.btnGETClick(Sender: TObject);
var
  LResponse: IResponse;
  DataSet: TMemDataset;
begin
  try
    DataSet := TMemDataset.Create(nil);
    LResponse := TRequest.New.BaseURL(edtBaseURL.Text)
      .Accept(edtAccept.Text)
      .DataSetAdapter(DataSet)
      .Get;

    ShowMessage(LResponse.Content);
    ShowMessage(DataSet.FieldByName('data').AsString);



  finally
    DataSet.Free;
  end;

  mmBody.Lines.Text := LResponse.Content;
  lblStatusCode.Caption := LResponse.StatusCode.ToString;
end;
